renderdoc/driver/gl/wrappers/gl_texture_funcs.cpp

```diff
@@ -1372,22 +1372,60 @@ void WrappedOpenGL::glCopyImageSubData(GLuint srcName, GLenum srcTarget, GLint s
       {
         TextureData &dstData = m_Textures[dstrecord->GetResourceID()];
 
-        if(srcX == 0 || srcY == 0 || srcZ == 0 || dstX == 0 || dstY == 0 || dstZ == 0)
-        {
-          // we only support whole copies - sub-copies will not work correctly.
-          RDCASSERT(srcWidth == RDCMAX(1, srcData.width >> srcLevel));
-          RDCASSERT(srcHeight == RDCMAX(1, srcData.height >> srcLevel));
-          RDCASSERT(srcDepth == RDCMAX(1, srcData.depth >> srcLevel));
+        GLsizei srcLevelWidth = RDCMAX(1, srcData.width >> srcLevel);
+        GLsizei srcLevelHeight = RDCMAX(1, srcData.height >> srcLevel);
+        GLsizei srcLevelDepth = RDCMAX(1, srcData.depth >> srcLevel);
 
-          RDCASSERT(srcWidth == RDCMAX(1, dstData.width >> dstLevel));
-          RDCASSERT(srcHeight == RDCMAX(1, dstData.height >> dstLevel));
-          RDCASSERT(srcDepth == RDCMAX(1, dstData.depth >> dstLevel));
+        GLsizei dstLevelWidth = RDCMAX(1, dstData.width >> dstLevel);
+        GLsizei dstLevelHeight = RDCMAX(1, dstData.height >> dstLevel);
+        GLsizei dstLevelDepth = RDCMAX(1, dstData.depth >> dstLevel);
```

Comment:
I think this calculation was wrong before. For 1D arrays, the `height` is the number of slices so it shouldn't be reduced by the mip level, and for other arrays (2D array, 2D MSAA array, cubemap array) the `depth` is the number of slices and similarly shouldn't be reduced.